src/initFile.ts

```diff
@@ -0,0 +1,76 @@
+import { ensureFile, join, resolve } from "./deps.ts";
+
+function isUrl(input: string): boolean {
+  try {
+    new URL(input);
+  } catch (_) {
+    return false;
+  }
+  return true;
+}
+
+function validateUrl(inputUrl: string | null): string {
+  if (!inputUrl || !isUrl(inputUrl.trim())) return "http://example.com";
```

Comment:
I think I would rather throw an error here or even better display an error message and ask again for a prompt